backend/src/router.py

```diff
@@ -12,26 +12,31 @@
 config = Config()
 
 
-def build_best_next_step_prompt(task, scratchpad):
+def build_agent_selection_system_prompt(task, scratchpad):
     agents_details = get_agent_details()
+    print(
+        "_______________________________________________________________________________________________________________________"
+    )
+    print("task", task)
+    print("list_of_agents", agents_details)
+    print("history", scratchpad)
     return prompt_engine.load_prompt(
-        "best-next-step",
-        task=json.dumps(task, indent=4),
+        "agent-selection-system-prompt",
         list_of_agents=json.dumps(agents_details, indent=4),
-        history=json.dumps(scratchpad, indent=4),
     )
 
 
-response_format_prompt = prompt_engine.load_prompt("agent-selection-format")
-
-
 async def build_plan(task, llm: LLM, scratchpad, model):
-    best_next_step_prompt = build_best_next_step_prompt(task, scratchpad)
+    agent_selection_system_prompt = build_agent_selection_system_prompt(task, scratchpad)
```

Comment:
Could bring the creation of the system prompt back in here, we don't often extract it out like this. If you are looking to keep the print stuff in, then keeping it in a separate function like `pretty_log` and update the `print`s to `log.info`s